tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -333,14 +332,20 @@ namespace tcf {
         return SIG_result;
     }
 
+    void WorkOrderProcessor::DecryptEncryptionKey(EnclaveData& enclaveData,
+		                          std::string encryptedSessionKey) {
+         ByteArray encryptedSessionKey_bytes = HexStringToBinary(encryptedSessionKey);
+         session_key = enclaveData.decrypt_message(encryptedSessionKey_bytes);
```

Comment:
Where are we handling decryption of "encryptedDataEncryptionKey" ? This function is handling decryption of encrypted session key alone.

encryptedDataEncryptionKey contains this encryption key in double encrypted format
First, it is encrypted with the worker's public encryption key (e.g. by a 3rd party that owns the data so the requester cannot see the data)
Then the result of the previous encryption above is encrypted with the key from encryptedSessionKey (by the requester to enforce the work order integrity)

Reply:
Currently we are not handling the encryption/decryption of encryptedDataEncryptionKey using using encryptedSessionKey at the client and enclave side. 
Can we take this up in the same PR or a seperate one?

Reply:
ok. Change param name encryptedSessionKey -> encryptedKey